### PR TITLE
Changes for the Demo

### PR DESCRIPTION
**Pull request by kanak8278**
payment plugin file name had typo, changed in backup.sql
Added short message of the last step completed in feedback. (`Feedback doesn't say what it did`)
added next line to the message in the NLR generation

### CODE REVIEW

**Comment by kanak8278** [comment]
This PR is connected to [Checker and Error Classification] (https://github.com/microsoft/PwR-NL2DSL/pull/11) PR in PwR-NL2DSL. Should be merged together.

Progress bar on PwR Studio working
image

Description of last step executed is also added in the Feedback.
image

File: app/jb.py [hunk 1]
```diff
@@ -184,15 +182,17 @@ def status_update_callback_wrapper(event, data):
         nl2dsl.nl2dsl()
 
         errors = nl2dsl.validate_dsl()
+
         nlr = generate_nlr(nl2dsl.dsl)
```
Comment (sameersegal) [inline app/jb.py:186]: We are generating nlr twice?

File: app/jb.py [hunk 2]
```diff
@@ -27,7 +26,7 @@ class Message(BaseModel):
 from nl2dsl import NL2DSL
 from .utils.codegen import CodeGen
 from .utils.nlr_gen import generate_nlr
-from .utils.feedback_gen import generate_feedback
+from .utils.feedback_gen import generate_feedback_v2
```
Comment (sameersegal) [inline app/jb.py]: Let's stick to only one version of generate_feedback. Remove the old one

File: app/jb.py [hunk 3]
```diff
@@ -184,15 +182,17 @@ def status_update_callback_wrapper(event, data):
         nl2dsl.nl2dsl()
 
         errors = nl2dsl.validate_dsl()
```
Comment (sameersegal) [inline app/jb.py]: Change the name of the variable since it carries errors, warnings, etc.

File: app/jb.py [hunk 4]
```diff
@@ -278,11 +278,9 @@ def status_update_callback(x):
             "optimizations": 0,
             "botExperience": "80%",
         }
-        # await self._progress(
-        #     Response(
-        #         # type="dsl_state",
-        #         message=json.dumps(program_state)
-        #     ))
+        await self._progress(
```
Comment (sameersegal) [inline app/jb.py:281]: Why is program_state hard coded just above this?

File: app/utils/feedback_gen.py [hunk 5]
```diff
@@ -1,7 +1,33 @@
 from .llm import llm, sm, um
 
 
-def generate_feedback(user_history, dsl_desc, errors, **kwargs):
+def generate_diff(old_nlr, new_nlr, **kwargs):
+    template_str = f"""You are a developer bot that is developing a program as per the following user instruction. 
+The program plan has been updated.
+Here is the older plan: {old_nlr}
```
Comment (sameersegal) [inline app/utils/feedback_gen.py:7]: Why are you not giving user instruction?

**Review by sameersegal — CHANGES_REQUESTED** [review]

File: app/jb.py (hunk 1 above)
Comment (kanak8278) [reply app/jb.py:186]: The two NLR are for diff generation. One is for input dsl, and another for the currently generated dsl.

File: app/utils/feedback_gen.py (hunk 5 above)
Comment (kanak8278) [reply app/utils/feedback_gen.py:7]: modified the user message prompt. We are giving both system and user message

File: app/jb.py (hunk 3 above)
Comment (kanak8278) [reply app/jb.py]: Changed `errors` into `issues`.

**Review by sameersegal — APPROVED** [review]

**Pull request merged**